File: js/arrays_strings/valid_mountain_array.js
const validMountainArray = (arr) => {
    let i=1;
    while(i<arr.length && arr[i]>arr[i-1]){
        i++;
    }
    if(i === 1 || arr.length === i){
        return false;
    }
    while(i<arr.length && arr[i]<arr[i-1]){
        i++;
    }
    return i === arr.length;
}

arr = [0,1,2,4,2,1]
console.log(validMountainArray(arr))

File: python/arrays_strings/valid_mountain_array.py
from typing import List


class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if(len(arr) < 3):
            return False

        i = 1
        while(i < len(arr) and arr[i] > arr[i-1]):
            i += 1

        if(i == 1 or i == len(arr)):
            return False

        while(i < len(arr) and arr[i] < arr[i-1]):
            i += 1

        return i == len(arr)


s = Solution()
arr = [1, 2, 3, 2, 1]
ans = s.validMountainArray(arr)
print(ans)
